_posts/en/newsletters/2021-07-21-newsletter.md

```diff
@@ -0,0 +1,110 @@
+---
+title: 'Bitcoin Optech Newsletter #158'
+permalink: /en/newsletters/2021/07/21/
+name: 2021-07-21-newsletter
+slug: 2021-07-21-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes recent changes to services and client
+software, discusses why wallets should wait before generating taproot
+addresses, lists new software releases and release candidates, and
+summarizes notable changes to popular Bitcoin infrastructure software.
+
+## News
+
+*No significant news this week.*
+
+## Changes to services and client software
+
+*In this monthly feature, we highlight interesting updates to Bitcoin
+wallets and services.*
+
+FIXME:bitschmidty
+
+## Preparing for taproot #5: why are we waiting?
+
+*A weekly [series][series preparing for taproot] about how developers
+and service providers can prepare for the upcoming activation of taproot
+at block height {{site.trb}}.*
+
+{% include specials/taproot/en/04-why-wait.md %}
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [Rust-Lightning 0.0.99][] is a release with a few API and
+  configuration changes.  See its [release notes][rl 0.0.99 rn] for
+  details.
+
+- [LND 0.13.1-beta.rc2][LND 0.13.1-beta] is a maintenance release with
+  minor improvements and bug fixes for features introduced in
+  0.13.0-beta.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
+repo], [Hardware Wallet Interface (HWI)][hwi repo],
+[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
+[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
+BOLTs][bolts repo].*
+
+- [Bitcoin Core #22112][] Force port 0 in I2P FIXME:jnewbery
+
+- [C-Lightning #4611][] updates the plugin-provided `keysend` RPC to
+  add a `routehints` parameter which allows providing information for
+  routing payments to [unannounced channels][topic unannounced
+  channels].
+
+- [C-Lightning #4646][] makes two changes in preparation for removing
+  old behavior.  The first change assumes all nodes support the
```

Comment:
```suggestion
  old behavior.  The first change assumes nodes support the
```

I don't think the 'all' is necessary here, and it jars a bit with the fact that nodes can signal non-support for TLV encoding.